File: trs-sdk-python/trs/exceptions.py
from __future__ import annotations


class TRSError(Exception):
    """Base SDK exception."""


class TRSConnectionError(TRSError):
    """Network or timeout failure."""


class TRSValidationError(TRSError):
    """Validation failure from trs-node or runtime verifier."""

    def __init__(self, message: str, *, errors: list[str] | None = None) -> None:
        super().__init__(message)
        self.errors = errors or []


class TRSServerError(TRSError):
    """Server-side failure."""


class TRSProtocolError(TRSError):
    """Unexpected response shape."""

File: trs-sdk-csharp/src/Trs.Sdk/Exceptions.cs
namespace Trs.Sdk;

public class TrsException : Exception
{
    public TrsException(string message, Exception? innerException = null) : base(message, innerException) { }
}

public sealed class TrsConnectionException : TrsException
{
    public TrsConnectionException(string message, Exception? innerException = null) : base(message, innerException) { }
}

public sealed class TrsValidationException : TrsException
{
    public IReadOnlyList<string> Errors { get; }

    public TrsValidationException(string message, IReadOnlyList<string> errors) : base(message)
    {
        Errors = errors;
    }
}

public sealed class TrsServerException : TrsException
{
    public TrsServerException(string message) : base(message) { }
}

public sealed class TrsProtocolException : TrsException
{
    public TrsProtocolException(string message) : base(message) { }
}

